File: commands/tournaments/signup.js
const fs = require('fs');
const path = require('path');
const { DATA_PATHS, WEBSITE_REPO, GITHUB_TOKEN } = require('../../utils/constants');
const { syncToSite } = require('../../utils/syncToSite');

const tournamentsPath = path.join(__dirname, '../../', DATA_PATHS.TOURNAMENTS);

module.exports = {
  name: 'signup',
  description: 'Sign up for a tournament',

  async execute(interaction) {
    const tournamentId = interaction.options.getString('tournament_id');
    const userId = interaction.user.id;

    let tournaments;
    try {
      tournaments = JSON.parse(fs.readFileSync(tournamentsPath, 'utf8'));
    } catch (err) {
      console.error('❌ Failed to read tournaments.json:', err);
      return interaction.reply({ content: 'Error reading tournament data.', ephemeral: true });
    }

    const tournament = tournaments.find(t => t.id === tournamentId);
    if (!tournament) {
      return interaction.reply({ content: 'Tournament not found.', ephemeral: true });
    }

    tournament.participants = tournament.participants || [];

    if (tournament.participants.includes(userId)) {
      return interaction.reply({
        content: 'You are already signed up!',
        ephemeral: true
      });
    }

    tournament.participants.push(userId);

    try {
      fs.writeFileSync(tournamentsPath, JSON.stringify(tournaments, null, 2));
      syncToSite('tournaments.json', WEBSITE_REPO, GITHUB_TOKEN);
    } catch (err) {
      console.error('❌ Failed to write tournaments.json:', err);
    }

    return interaction.reply({
      content: `🎉 You have joined **${tournament.name}**!`,
      ephemeral: true
    });
  }
};

File: commands/tournaments/signup.py
# signup.py
import json
import discord
from discord import app_commands
from discord.ext import commands
from utils.constants import DATA_PATHS, WEBSITE_REPO, GITHUB_TOKEN
from utils.syncToSite import sync_to_site

TOURNAMENTS_PATH = DATA_PATHS["TOURNAMENTS"]

class Signup(commands.Cog):
    def __init__(self, bot: commands.Bot):
        self.bot = bot

    @app_commands.command(
        name="signup",
        description="Sign up for a tournament"
    )
    @app_commands.describe(tournament_id="The ID of the tournament")
    async def signup(self, interaction: discord.Interaction, tournament_id: str):
        user_id = str(interaction.user.id)

        try:
            with open(TOURNAMENTS_PATH, "r", encoding="utf8") as f:
                tournaments = json.load(f)
        except Exception as e:
            print(f"❌ Failed to read tournaments.json: {e}")
            return await interaction.response.send_message("Error reading tournament data.", ephemeral=True)

        tournament = next((t for t in tournaments if t.get("id") == tournament_id), None)
        if not tournament:
            return await interaction.response.send_message("Tournament not found.", ephemeral=True)

        tournament.setdefault("participants", [])
        if user_id in tournament["participants"]:
            return await interaction.response.send_message("You are already signed up!", ephemeral=True)

        tournament["participants"].append(user_id)

        try:
            with open(TOURNAMENTS_PATH, "w", encoding="utf8") as f:
                json.dump(tournaments, f, indent=2)
            sync_to_site("tournaments.json", WEBSITE_REPO, GITHUB_TOKEN)
        except Exception as e:
            print(f"❌ Failed to write tournaments.json: {e}")

        await interaction.response.send_message(f"🎉 You have joined **{tournament.get('name')}**!", ephemeral=True)

async def setup(bot: commands.Bot):
    await bot.add_cog(Signup(bot))

